src/include/wiredtiger.in

```diff
@@ -2173,6 +2195,52 @@ struct __wt_connection {
 	/*! @} */
 
 	/*!
+	 * @name Transactions
+	 * @{
+	 */
+	/*!
+	 * Query the global transaction timestamp state.
+	 *
+	 * @snippet ex_all.c query timestamp
+	 *
+	 * If there is no matching timestamp (e.g., if this method is called
+	 * before timestamps are used) ::WT_NOTFOUND will be returned.
+	 *
+	 * @param connection the connection handle
+	 * @param[out] hex_timestamp a buffer that will be set to the
+	 * hexadecimal encoding of the timestamp being queried.  Must be large
+	 * enough to hold a hex-encoded timestamp (i.e., double the timestamp
+	 * size plus one byte for NUL termination).
+	 * @configstart{WT_CONNECTION.query_timestamp, see dist/api_data.py}
+	 * @config{get, specify which timestamp to query: \c all_committed
+	 * returns the largest timestamp such that all earlier timestamps have
+	 * committed.  See @ref transaction_timestamps., a string\, chosen from
+	 * the following options: \c "all_committed"; default \c all_committed.}
+	 * @configend
+	 * @errors
+	 */
+	int __F(query_timestamp)(
+	    WT_CONNECTION *connection, char *hex_timestamp, const char *config);
+
+	/*!
+	 * Set a global transaction timestamp.
+	 *
+	 * @snippet ex_all.c set oldest timestamp
+	 *
+	 * @param connection the connection handle
+	 * @configstart{WT_CONNECTION.set_timestamp, see dist/api_data.py}
+	 * @config{oldest_timestamp, future commits and queries will be no
+	 * earlier than the specified timestamp.  Supplied values must be
+	 * monotonically increasing.  see @ref transaction_timestamps., a
+	 * string; default empty.}
+	 * @configend
+	 * @errors
+	 */
+	int __F(set_timestamp)(
+	    WT_CONNECTION *connection, const char *config);
+	/*! @} */
```

Comment:
I deliberately changed from `set_oldest_timestamp` to `set_timestamp("oldest_timestamp=X")` so that we have the flexibility to add future timestamp information.  For example, I expect to reuse this function in the recover to a timestamp project for `set_timestamp("durable_timestamp=X")`